packages/core/content-manager/server/src/preview/controllers/validation/preview.ts

```diff
@@ -0,0 +1,53 @@
+import * as yup from 'yup';
+import { pick } from 'lodash/fp';
+
+import type { Core, UID } from '@strapi/types';
+import { validateYupSchema, errors } from '@strapi/utils';
+
+import { Preview } from '../../../../../shared/contracts';
+import type { HandlerParams } from '../../services/preview-config';
+
+const getPreviewUrlSchema = yup
+  .object()
+  .shape({
+    documentId: yup.string(),
+    locale: yup.string().nullable(),
+    status: yup.string(),
```

Comment:
haven't manually tried, but arent fields optional by default in yup?. I know its the opposite of zod 😄 but I get confused between the too

Reply:
you're right, `required()` exists, so it is indeed optional by default. So in that case, shouldn't documentId be required?

Reply:
on single types it will be undefined , and there is this additional check https://github.com/strapi/strapi/pull/21574/files#diff-9be1cdb997d59f9cff5687dfa8b3b802bd113aed5824633ffed5235b541086a8R37

Reply:
I will add a note!

Reply:
Yup I just read your code that explained this, sorry I'm not fully awake yet apparently 😁

Reply:
😆 mondays gonna monday
![image](https://github.com/user-attachments/assets/9e5219de-1ab2-4298-92ff-64325ff6d244)